blog/_posts/2017-02-25-scalafix-v0.3.md

```diff
@@ -0,0 +1,158 @@
+---
+layout: blog
+post-type: blog
+by: Ólafur Páll Geirsson
+title: "Refactor with scalafix v0.3"
+---
+
+I am happy to announce the release of [scalafix v0.3][scalafix], a library and tool to rewrite Scala source code.
+Scalafix is developed at the [Scala Center][sc] with the long-term mission to help automate the migration of between Scala versions.
+However, as I hope to demonstrate in this post, scalafix can be used for more than just migrating between Scala versions.
+Scalafix can also be used for ad-hoc library and application migrations.
+
+Scalafix v0.3 uses the new scala.meta semantic API to provide a re-designed `Rewrite` and `Patch` API to implement custom refactorings.
+Let me explain what that means word by word.
+
+> Note. This is the second post on scalafix and scala.meta. You might be
+> interested in reading the first post
+> [here](http://scala-lang.org/blog/2016/10/24/scalafix.html).
+
+## Scala.meta semantic API
+
+Scala.meta recently announced the [first release of its semantic API][meta-1.6].
+This release is the product of close collaboration for the past several months between
+[@xeno_by][] at Twitter and [myself][@olafurpg] at the Scala Center.
+The objective of the semantic API is to provide operations to query information from the compiler.
+
+The first version of the scala.meta semantic API makes it possible to query for the resolved "symbol" of a name that appears in a Scala source file.
+A name is a reference to some definition, for example `println` or `scala.Predef.println`.
+A symbol is a unique identifier of a single definition.
+For example, `println` from the standard library has the symbol `_root_.scala.Predef.println(Ljava/lang/Object;)V.`.
+The compiler is responsible for resolving names to symbols.
+
+[Scalahost][] is a compiler plugin in the scala.meta project that extracts symbols from the compiler and maps them to scala.meta syntax trees.
```

Comment:
Cudos to @dwijnand for teaching me [here](https://github.com/olafurpg/scala-lang/pull/1#discussion_r103059014) 🌟